untrusted/quote_verify/src/lib.rs

```diff
@@ -103,6 +129,40 @@ impl Quote {
         self.verify_signature(bytes, QUOTING_ENCLAVE_SIGNATURE_START, key)
     }
 
+    /// Verify the attestation key in the quote is valid.
+    pub fn verify_attestation_key(&self) -> Result<(), Error> {
+        let mut hasher = Sha256::new();
+
+        let key = &self.bytes[ATTESTATION_KEY_START..ATTESTATION_KEY_START + KEY_SIZE];
+        hasher.update(key);
+
+        let authentication_data = self.get_qe_authentication_data();
+        hasher.update(authentication_data);
+
+        let hash = hasher.finalize();
+        let report_data = &self.bytes
+            [QUOTING_ENCLAVE_REPORT_DATA_START..QUOTING_ENCLAVE_REPORT_DATA_START + hash.len()];
+
```

Comment:
Is there any reason to also verify that the last 32 bytes of the Report Data are 0x00's? It shouldn't be relevant to the verification of the attestation key but it is described in Table 4.

Reply:
Add logic and tests for this.

I'm a bit torn if this is perhaps making the check too strict in the event Intel wants to use those 32 bytes for something.  Since we're comparing a SHA256 digest, it seems unlikely we would get a valid digest and unexpected values in there.
But I didn't feel strongly enough to push back on it.